presto-main/src/main/java/com/facebook/presto/metadata/InternalNode.java

```diff
@@ -66,24 +66,32 @@ public int getStatusCode()
     private final boolean coordinator;
     private final boolean resourceManager;
     private final NodeStatus nodeStatus;
+    //the pool that this node belongs to
+    private final String pool;
 
     public InternalNode(String nodeIdentifier, URI internalUri, NodeVersion nodeVersion, boolean coordinator)
     {
-        this(nodeIdentifier, internalUri, nodeVersion, coordinator, false);
+        this(nodeIdentifier, internalUri, nodeVersion, coordinator, false, "general");
```

Comment:
"general" is a [magic string](https://deviq.com/antipatterns/magic-strings).
Can you add this as a constant somewhere and refer it all the places we are using it.

Reply:
"general" seems a bit ambiguous, maybe something "default" or "default_pool"?

I'm mostly curious how the other non-general pools will be specified? i.e. how will pool concept expose/utilized by coordinator/users?

Reply:
I was thinking of using the constant com.facebook.airlift.discovery.client.ServiceSelectorConfig.[DEFAULT_POOL](https://github.com/prestodb/airlift/blob/master/discovery/src/main/java/com/facebook/airlift/discovery/client/ServiceSelectorConfig.java#L24), 
The value of this constant is "general" which will be in sync with default pool used for [NodeInfo](https://github.com/prestodb/airlift/blob/master/node/src/main/java/com/facebook/airlift/node/NodeConfig.java#L36) 
wdyt @ajaygeorge @chliang71 ? 

Regarding the usage of pool, we have this WIP PR https://github.com/prestodb/presto/pull/16720 , where we are planning to filter nodes during scheduling based on pools.